lib/instances/ScreenGui.lua

```diff
@@ -19,17 +19,21 @@ ScreenGui.properties.AbsoluteSize = InstanceProperty.readOnly({
 	end,
 })
 
-ScreenGui.properties.DisplayOrder = InstanceProperty.normal({
-	set = function(self, key, value)
-		if type(value) ~= "number" then
-			error(string.format("%s must be a number", key), 2)
-		end
+ScreenGui.properties.DisplayOrder = InstanceProperty.typed("number", {
+	getDefault = function()
+		return 0
+	end,
+})
 
-		getmetatable(self).instance.properties[key] = value
+ScreenGui.properties.AutoLocalize = InstanceProperty.typed("boolean", {
+	getDefault = function()
+		return true
 	end,
+})
```

Comment:
I guess we don't have `GuiBase` or `GuiBase2d` yet, but this property really belongs there.

Reply:
Oh yeah, I didn't realize that. Should I make GuiBase2d and pull it out?

Reply:
Actually, `AbsoluteSize` and `AbsolutePosition` are from `GuiBase2d` as well. And `ZIndexBehavior` is from `LayerCollector`...